src/scripts/main.js

```diff
@@ -1,3 +1,68 @@
 'use strict';
 
 // write code here
+
+const rowAppended = document.querySelector('.append-row');
+const rowRemover = document.querySelector('.remove-row');
+const columnAppended = document.querySelector('.append-column');
+const columnRemover = document.querySelector('.remove-column');
+const field = document.querySelector('.field');
+
+function updateButtonStates() {
+  const trs = field.querySelectorAll('tr');
+  const tds = trs[0] ? trs[0].children : [];
+
+  rowAppended.disabled = trs.length >= 10;
+
+  rowRemover.disabled = trs.length <= 2;
+
+  columnAppended.disabled = tds.length >= 10;
+
+  columnRemover.disabled = tds.length <= 2;
+}
+
+updateButtonStates();
+
+rowAppended.addEventListener('click', (e) => {
+  const trs = document.querySelectorAll('tr');
+
+  if (trs.length < 10) {
+    const trCopy = field.querySelector('tr').cloneNode(true);
```

Comment:
Cloning the first row to create a new row might not be ideal if the first row contains specific content or attributes. Consider creating a new row with empty cells instead to avoid duplicating content.